### DISCUSSION TITLE
Issue with Installing docker-magento for exixting project  on Linux.

### DISCUSSION THREAD

**Discussion by PrashantPunter** [markshust/docker-magento#1031] [General]
I am trying to install docker-Magento for an Existing Project. So I added  

` phpfpm:
    image: markoshust/magento-php:8.2-fpm-0
    volumes: *appvolumes
    env_file: env/phpfpm.env
    extra_hosts: "host.docker.internal:172.17.0.1"`

Then when I am running:

`bin/start --no-dev`

I am getting error:
`validating /home/user/Sites/magento2.4.7-t2/compose.yaml: services.phpfpm.extra_hosts must be a mapping`

**Comment by YevhenZvieriev** [comment]
Hi, @darkpasanger 

The extra_hosts parameter must be defined as a map rather than a string. 

Replace the line:

```yaml
extra_hosts: "host.docker.internal:172.17.0.1"
```

on the :

```yaml
extra_hosts:

- "host.docker.internal:172.17.0.1"
```

Also, I recommend adding extra_hosts to each service under the app and phpfpm keys.

Your compose.yaml file should look something like this:
![Screenshot_2023-12-28_02-11-33](https://github.com/markshust/docker-magento/assets/43544955/b23188b9-9b1f-4874-86ee-3ee71cb632ce)

Save the changes and try to start the containers again. This should fix the error you received.

Feel free to let me know if you have further questions or need more assistance!

**Comment by PrashantPunter** [comment]
Thanks for the help. As mentioned I copied the MySQL creds from `src/app/etc/env.php` to db.env. But when running 
``bin/mysql < ../existing/magento.sql`
getting error.

`ERROR 2002 (HY000): Can't connect to MySQL server on '127.0.0.1' (115)`

`

**Comment by YevhenZvieriev** [comment]
Hi, @darkpasanger 

May I know your creds?

Also, you can try this way, I hope it will help you :

```code
bin/mysql -uroot -pmagento <  ../existing/magento.sql
```

**Comment by PrashantPunter** [comment]
From `src/app/etc/env.php`
```
      array (
        'host' => 'localhost',
        'dbname' => 'betgdsqmyy',
        'username' => 'betgdsqmyy',
        'password' => 'nqu8Qck8tf',
        'model' => 'mysql4',
        'engine' => 'innodb',
        'initStatements' => 'SET NAMES utf8;',
        'active' => '1',
      ),
```
and my db.env look likes
```
MYSQL_HOST=127.0.0.1
MYSQL_ROOT_PASSWORD=nqu8Qck8tf
MYSQL_DATABASE=betgdsqmyy
MYSQL_USER=betgdsqmyy
MYSQL_PASSWORD=nqu8Qck8tf

MYSQL_INTEGRATION_ROOT_PASSWORD=nqu8Qck8tf
MYSQL_INTEGRATION_DATABASE=magento_integration_tests
MYSQL_INTEGRATION_USER=magento
MYSQL_INTEGRATION_PASSWORD=magento
MYSQL_INTEGRATION_HOST=db
```
I try using localhost as MYSQL_HOST=loaclhost but it was throwing the error 
`ERROR 1227 (42000) at line 15423: Access denied; you need (at least one of) the SUPER, SET USER privilege(s) for this operation
`

**Comment by YevhenZvieriev** [comment]
Ok, I got it!

1.  You made a mistake, you should  copy the MySQL creds from `db.env` to `src/app/etc/env.php`

2. In your `db.env` file, you use the incorrect host address for MySQL. 
You should use  `db`  instead of `127.0.0.1` 

3.  Update `app/etc/env.php` :
Fix the error in the `app/etc/env.php` file. You have an error in the  `'host'` configuration. Instead of `'localhost'`, you should use `'db'` to point to the database container.
If you leave the value  `'localhost'` in the line `'host' => 'localhost'`, then Magento will try to connect to the database that is located in the same container as Magento itself. However, in your case, the database is hosted on a different container, which is marked as `'db'`. Open the `docker-compose.yml` file and make sure.

Replace  `'host' => 'localhost'`   with  `'host' => 'db'`

Your creds in `env.php` should look something like this:
```code
   'db' => [
        'table_prefix' => '',
        'connection' => [
            'default' => [
                'host' => 'db',
                'dbname' => 'magento',
                'username' => 'magento',
                'password' => 'magento',
                'model' => 'mysql4',
                'engine' => 'innodb',
                'initStatements' => 'SET NAMES utf8;',
                'active' => '1',
                'driver_options' => [
                    1014 => false
                ]
            ]
        ]
    ],
```
4. Your `db.env` file should look something like this:

```
MYSQL_HOST=db
MYSQL_ROOT_PASSWORD=magento
MYSQL_DATABASE=magento
MYSQL_USER=magento
MYSQL_PASSWORD=magento

MYSQL_INTEGRATION_ROOT_PASSWORD=magento
MYSQL_INTEGRATION_DATABASE=magento_integration_tests
MYSQL_INTEGRATION_USER=magento
MYSQL_INTEGRATION_PASSWORD=magento
MYSQL_INTEGRATION_HOST=db
```

5.  `bin/copytocontainer --all` (to synchronize env.php to the container)

6. Restart the containers

7. `bin/mysql < ../existing/magento.sql`
  
If you get an error, use this command:

`bin/mysql -uroot -pmagento < ../existing/magento.sql`

I also had this error: ` ERROR 1227 (42000) at line 15423: Access denied; you need (at least one of) the SUPER, SET USER privilege(s) for this operation  `

I resolved this next way: 
`bin/clinotty mysql -hdb -uroot -pmagento magento < ../existing/magento.sql`

8. To confirm that the existing database was imported you can do this:
`bin/mysql -e "EXPLAIN core_config_data"`
